library.lua

```diff
@@ -154,6 +154,26 @@ function _.filter(t, p)
   return out
 end
 
+function _.sample_size_u(t, n)
+  _.expect('sample_size_u', 1, 'table', t)
+  _.expect('sample_size_u', 2, 'number', n)
+
+  if #t <= n then
+    return t
+  end
+
+  local src = _.keys(t)
+  local out = {}
+  for i = 1, n do
+    local k = _.sample(src)
+    out[i] = t[k]
+
+    src[k] = src[#src]
+    src[#src] = nil
```

Comment:
Can't you use table.remove?

Reply:
This is more efficient than table.remove since we don't care about ordering, because table.remove will do `t[k] = nil; t[k] = t[k+1]; t[k+1] = t[k+2]......`

Reply:
Ah, fair enough. TIL